### DISCUSSION TITLE
Get multiple borders of black/white image.

### DISCUSSION THREAD

**Discussion by petrasvestartas** [SixLabors/ImageSharp#1231] [Q&A] [answered]
Hi,

I am new to ImageSharp, I was using Accord.NET, Aforge but these libraries became outdated.

I am searching for a method to get multiple borders of  a Bitmap that contains white shapes.

I would like to get not one but all of possible borders:
![border-following](https://user-images.githubusercontent.com/18013985/84683599-bd2b1200-af37-11ea-9891-322ee9a77cd9.png)

**Comment by JimBobSquarePants** [comment, marked as answer]
I'm afraid no methods like this exist. 

ImageSharp isn't a computer vision library (although we've thrown in a few useful methods). Rather it provides the tools to build a computer vision library upon. 

In theory, porting methods to ImageSharp should be fairly trivial as we provide simplified per-pixel access and remove some of the complexity regarding dealing with pixels formats. I would have a look at the source of what you need and have a go at implementing a port. If you have any questions regarding our API please let us know.

**Reply by antonfirsov** [reply]
I think the use case is not fully clear from the description. @petrasvestartas is this about pixelgraphic edge detection, or dou you also need to extract vectorgraphic shapes for the edges?
